test/CMakeLists.txt

```diff
@@ -127,19 +126,24 @@ set_source_files_properties(test_query_geo.cpp PROPERTIES
     COMPILE_FLAGS "$<$<CXX_COMPILER_ID:GNU>: -Wno-unknown-pragmas>"
 )
 
-# FIXME: Benchmarks
-
 if (MSVC)
     append_source_file_compile_options(FILES test_query.cpp test_query_big.cpp FLAGS /bigobj)
 endif()
 
 # Resources required for running the tests
-file(GLOB REQUIRED_TEST_FILES RELATIVE ${CMAKE_CURRENT_BINARY_DIR}
+file(GLOB REQUIRED_TEST_FILES
      "*.json"
      "*.realm"
      "expect_string.txt")
 
-add_executable(CoreTests main.cpp ${CORE_TESTS} ${REQUIRED_TEST_FILES} ${REALM_TEST_HEADERS})
+add_library(CoreTestLib OBJECT ${CORE_TESTS} ${REQUIRED_TEST_FILES} ${REALM_TEST_HEADERS})
+enable_stdfilesystem(CoreTestLib)
+target_link_libraries(CoreTestLib Storage QueryParser)
+
+add_executable(CoreTests main.cpp test_all.cpp ${REQUIRED_TEST_FILES})
+target_sources(CoreTests PUBLIC $<TARGET_OBJECTS:CoreTestLib>)
+target_sources(CoreTests PUBLIC $<TARGET_OBJECTS:TestUtil>)
```

Comment:
Can't we link against `TestUtil` as a normal library instead of just including the object files directly?

Reply:
Yeah, my understanding is that object libraries are just a build time optimization for when the archive file is large enough that creating it takes a meaningful amount of time and space, but TestUtil *hopefully* is very small. If we're having to work around any of the object library quirks then we should probably just use a static library?

Reply:
I reverted TestUtil back to a static library.